src/Internal/RejectedPromise.php

```diff
@@ -15,18 +16,54 @@
 final class RejectedPromise implements PromiseInterface
 {
     private $reason;
+    private $handled = false;
 
     public function __construct(\Throwable $reason)
     {
         $this->reason = $reason;
     }
 
+    public function __destruct()
+    {
+        if ($this->handled) {
+            return;
+        }
+
+        $message = 'Unhandled promise rejection with ';
+
+        if ($this->reason instanceof Throwable) {
+            $message .= get_class($this->reason) . ': ' . $this->reason->getMessage();
+            $message .= ' raised in ' . $this->reason->getFile() . ' on line ' . $this->reason->getLine();
+            $message .= PHP_EOL . $this->reason->getTraceAsString();
+        } else {
+            if ($this->reason === null) {
+                $message .= 'null';
+            } else {
+                $message .= (is_object($this->reason) ? get_class($this->reason) : gettype($this->reason));
+            }
+
+            $trace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+            if (isset($trace[0]['file'], $trace[0]['line'])) {
+                $message .= ' detected in ' . $trace[0]['file'] . ' on line ' . $trace[0]['line'];
+            }
+
+            ob_start();
+            debug_print_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+            $message .= PHP_EOL . ob_get_clean();
+        }
+
+        $message .= PHP_EOL;
+        fatalError($message);
+    }
```

Comment:
As `$this->reason` is always a `\Throwable` , the `else` block can be removed.

Reply:
@jsor ah yes good point, totally forgot to check that after a fail rebase and doing everything by hand :joy: . Just pushed an update